core/src/main/scala/quasar/plugin/jdbc/TransactorConfig.scala

```diff
@@ -30,19 +30,25 @@ import scala.concurrent.duration._
   *                              idle connections exceeding their lifetime will be replaced with
   *                              new ones. Make sure to set this several seconds shorter than any
   *                              database or infrastructure imposed connection time limit. (default: 30min)
+  * @param connectionInitFailTimeout the time before the pool initialization fails, or 0 to
```

Comment:
I don't think we want to leak so many of the particular pool semantics here, `Option[FiniteDuration]` where `None` means "don't validate" is a bit more explicit, a sum type would be even better. If we want to prevent the `0` behavior with Hikari, we can validate and/or document that the duration should be positive.

Reply:
Right, I thought and hesitated about it but went for this since `FiniteDuration` can also be negative. 
So `FiniteDuration` is also not a great fit, since at one side it needs to be tightened by the validation and at the other side it needs to be widened.

I'm personally inclined to not trying to stop these kind of leaks that are introduced by 3rd party libs, unless it's trivial. If you really don't like this I can change it though.

Reply:
This config type is intentionally generic so that we have control over what semantics we're exposing to connectors that use it, in this case we're choosing to expose more than we need to.

Another option would be to just disable the validation here, for now, then you wouldn't have to modify any of the use-sites.

Reply:
Would the sum type
```scala
sealed trait ConnectionInit

case object ValidateConnection extends ConnectionInit
case object SkipValidateConnection extends ConnectionInit
final case class TimeoutConnection(d: FiniteDuration) extends ConnectionInit
```

cover all our needs?

Reply:
I think so. @wemrysi are you ok with this too?

Reply:
I think a binary "validate on start" | "don't validate on start" would be preferable (the `initFailTimeout` isn't very intuitive as it is an additional timeout applied after the connection timeout when initializing the pool), but I'm fine with either.

Reply:
Ah right that's true. I'll go with that if there are no objections.

Reply:
Ah, so we don't actually need to timeout the connection. The goal is to either validate or not validate. 

```scala
sealed trait PoolInit

case object ValidateOnStart extends PoolInit
case object DoNotValidateOnStart extends PoolInit
```
In this case, `ValidateOnStart` would become `c.setInitializationFailTimeout(1)`, and `DoNotValidateOnStart` would become `c.setInitializationFailTimeout(-1)`.